### DISCUSSION TITLE
BIRT prevent Page Break

### DISCUSSION THREAD

**Discussion by tnation1424** [eclipse-birt/birt#1500] [Q&A]
In the image I posted notice how there is a big gap of white space on the first page and then it picks up on with the paragraph on the next page. I realized BIRT is doing a new page because it won't be able fit the paragraph on previous page.  My question is there way of having BIRT display part of the paragraph on the first page, and then the remain content on the next page?

![image](https://github.com/eclipse-birt/birt/assets/149100941/4fa76aa4-3fae-421c-9f8c-10d3aeb7076e)

**Comment by stevehydro** [comment]
check your grid/table Page Break.  Where it says "Inside - Auto", set it from "Auto" to "Avoid".

See if that helps.  I have lots of issues with this as well.

**Reply by hvbtup** [reply]
That's the opposite effect. It forces BIRT to start on a new page if the text doesn't fit completely in the current page.

**Reply by stevehydro** [reply]
oh right, it will keep whatever is inside together.

**Comment by tnation1424** [comment]
Thank you for the suggestion, but it had no effect.   I've tried everything the Page Breaks and no success.  You mention you had lots of issues as well.  When you did have success what was thing that was the fix?

**Comment by hvbtup** [comment]
If your paragraph is a dynamic text item with HTML content, then you might hit a bug: https://github.com/eclipse-birt/birt/issues/1443

Note: The page break property `page-break-before` must be `always`, otherwise the text will always start on a new page, and `page-break-inside` must not be `avoid`, otherwise the text will not start on the current page if it doesn't fit completely.

**Comment by tnation1424** [comment]
Thank you so much for the information!  Now I know what is causing the problem.  The report does use dynamic Text with HTML content.  I will rewrite that section of the report.  Again thank you.
